apps/backend/src/tests/response_api.test.ts

```diff
@@ -0,0 +1,555 @@
+      'Content and/or status value is invalid.',
+    );
+
+    const responseInDbAfterUpdate = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+
+    expect(responseInDbAfterUpdate).toEqual(responseInDbBeforeUpdate);
+  });
+
+  test("Empty input doesn't change anything on the server", async () => {
+    const responseInDbBeforeUpdate = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+
+    const httpResponse: Response = await api
+      .put(`/api/responses/${ANTONINAS_RESPONSE_ID}`)
+      .set('Authorization', `Bearer ${token}`)
+      .send({});
+
+    const responseInDbAfterUpdate = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+
+    expect(httpResponse.body.id).toBe(ANTONINAS_RESPONSE_ID);
+    expect(responseInDbAfterUpdate).toEqual(responseInDbBeforeUpdate);
+    expect(httpResponse.status).toEqual(200);
+  });
+
+  test('id, user_id, request_id and time_created fields cannot be edited', async () => {
+    // Cheking for failure to update id
+    const response1 = await api
+      .put(`/api/responses/${ANTONINAS_RESPONSE_ID}`)
+      .set('Authorization', `Bearer ${token}`)
+      .send({ id: 1000 });
+
+    expect(response1.status).toBe(400);
+    expect(response1.body.error).toBe(
+      'Content and/or status value is invalid.',
+    );
+
+    // Checking for failure to update user_id
+    const response2 = await api
+      .put(`/api/responses/${ANTONINAS_RESPONSE_ID}`)
+      .set('Authorization', `Bearer ${token}`)
+      .send({ user_id: 1000 });
+
+    expect(response2.status).toBe(400);
+    expect(response2.body.error).toBe(
+      'Content and/or status value is invalid.',
+    );
+
+    // Checking for failure to update request_id
+    const response3 = await api
+      .put(`/api/responses/${ANTONINAS_RESPONSE_ID}`)
+      .set('Authorization', `Bearer ${token}`)
+      .send({ request_id: 1000 });
+
+    expect(response3.status).toBe(400);
+    expect(response3.body.error).toBe(
+      'Content and/or status value is invalid.',
+    );
+
+    // Checking for failure to update time_created
+    const response4 = await api
+      .put(`/api/responses/${ANTONINAS_RESPONSE_ID}`)
+      .set('Authorization', `Bearer ${token}`)
+      .send({ time_created: new Date() });
+
+    expect(response4.status).toBe(400);
+    expect(response4.body.error).toBe(
+      'Content and/or status value is invalid.',
+    );
+
+    const responseInDbAfterUpdate = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+    expect(responseInDbAfterUpdate).toEqual(responseInDbAfterUpdate);
+  });
+
+  test('Able to change to status of response to ACCEPTED', async () => {
+    const initialResponseInDb = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+    expect(initialResponseInDb.status).toBe('PENDING');
+
+    // Why can we not use type of `updateData` to be `UpdateRequestData` ?
```

Comment:
This has changed on my end. I think you have to generate new prisma migration to get the types to updated based on the latest types

Reply:
done.